File: src/hackerrank/practice/MinimumLoss.java
package hackerrank.practice;

import java.util.*;

public class MinimumLoss {
    public static int minimumLoss(List<Long> price) {
        // Write your code here

        Map<Long, Integer> map = new HashMap<>();

        for (int i = 0; i < price.size(); i++) {
            map.put(price.get(i), i);
        }

        Collections.sort(price);

        int ans = Integer.MAX_VALUE;

        for (int i = 0; i < price.size() - 1; i++) {
            if (map.get(price.get(i)) > map.get(price.get(i + 1))) {

                int diff = (int) (price.get(i + 1) - price.get(i));

                if (diff > 0 && diff < ans) {
                    ans = diff;
                }
            }
        }
        return ans;

    }
}

File: src/hackerrank/practice/MinimumLoss.kt
package hackerrank.practice

import java.util.*

fun minimumLoss(price: Array<Long>): Int {

    val sortedArray = price.sortedArray()

    var ans = Long.MAX_VALUE

    val map = price.mapIndexed { i, v ->
        v to i
    }.toMap()

    for (i in 0 until price.size - 1) {
        if (map[sortedArray[i]]!! > map[sortedArray[i + 1]]!!) {
            ans = Math.min(ans, (sortedArray[i + 1] - sortedArray[i]))
        }
    }
    return ans.toInt()
}

fun main(args: Array<String>) {

    val sc = Scanner(System.`in`)
    val n = sc.nextInt()

    val price = Array(n) {
        sc.nextLong()
    }

    sc.close()

    val result = minimumLoss(price)

    println(result)
}
